src/app.rs

```diff
@@ -2542,9 +2544,25 @@ impl UiExtra for egui::Ui {
 pub fn start(data_sources: Vec<Box<dyn DeferredDataSource>>) {
     env_logger::try_init().unwrap_or(()); // Log to stderr (if you run with `RUST_LOG=debug`).
 
+    let all_locators = data_sources.iter().fold(Vec::new(), |acc, x| {
+        acc.into_iter()
+            .chain(x.fetch_description().source_locator)
```

Comment:
This seems like not the right way to write this. I'm pretty sure flatten would be shorter and more efficient.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flatten

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/20ba2e8e877f7d44a34eb35958ed0eb07334d635